ldap_probe/models.py

```diff
@@ -358,6 +362,54 @@ def node(self):
 
         return self.ad_node.get_node()
 
+    @property
+    def perf_alert(self):
+        """
+        flag for considering if an instance of this class must trigger a
+        performance alert
+
+        :returns: `True/False`
+        :rtype: bool
+        """
+        return any(
+            [
+                elapsed for elapsed in
+                [
+                    self_elapsed for self_elapsed in
+                    [
+                        self.elapsed_bind, self.elapsed_anon_bind,
+                        self.elapsed_search_ext, self.elapsed_read_root
+                    ]
+                    if self_elapsed is not None
+                ]
+                if elapsed >= get_preference('ldapprobe__ldap_perf_alert')
+            ]
+        )
+
+    @property
+    def perf_warn(self):
```

Comment:
To avoid duplication of code create a helper function 'perf_helper' (or 'perf_for_limit' or whatever name you think is most descriptive), that you can pass the preference into.
That is, perf_warn would become `return self.perf_helper(get_preference('ldapprobe__ldap_perf_warn'))` and perf_alert would be `return self.perf_helper(get_preference('ldapprobe__ldap_perf_alert'))`.

Reply:
the idea behind using ```@property``` decorators is to avoid having to pass  extra arguments to the ```signals.invoke_raise_ldap_perf_alert()``` function. it is already a messy function to call so we want to keep it as simple as possible.
the other reason is rather obscure. if one tries to import get_preference into a signals module (which is imported the django way in the apps module), you will a circular import error. i had that one bite me in the butt a few months ago.